### PR TITLE
post: ng v13

### PR DESCRIPTION
**Pull request by cexbrayat**

### CODE REVIEW

File: _posts/2021-11-03-angular-cli-13.0.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    ng update @angular/core@12 @angular/cli@12
+    # check that everything is ok
+    # then
+    ng update @angular/core@13 @angular/cli@13
+
+
+## Deprecations
+
+The `deployUrl` option for `ng build` has been deprecated.
+It was previously introduced to handle applications that only assets are hosted on a CDN.
```
Comment (jnizet) [inline _posts/2021-11-03-angular-cli-13.0.md:75]: to handle applications that only store their assets on a CDN

File: _posts/2021-11-03-angular-cli-13.0.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+## ng serve
+
+The dev server is now using webpack-dev-server v4,
+which means it relies on WebSockets to communicate changes to the browser
+to trigger a live-reload.
+If you're using a proxy, remember to enable proxy on WebSockets as well.
+
+As we're talking about the proxy configuration,
+note that it is now possible to define the configuration as an ESM module (ending with `.mjs`),
```
Comment (jnizet) [inline _posts/2021-11-03-angular-cli-13.0.md:89]: To define its configuration

File: _posts/2021-11-03-angular-cli-13.0.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+in addition to JSON and CommonJS:
+
+    export default {
+      '/api/*': { target: 'http://localhost:8080/api" }
+    }
+
+A ton of work has been done to improve the ESM support in the CLI and the framework
+Angular now ships as an ESM as well,
```
Comment (jnizet) [inline _posts/2021-11-03-angular-cli-13.0.md:97]: comma -> dot.

File: _posts/2021-11-03-what-is-new-angular-13.0.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+a limitation of TypeScript which did not allow a setter to have a different type than its getter.
+So until now, if you wanted a component to accept a boolean or a string for an `@Input()` setter, and the getter to only return a boolean, you had to use the weird "input type coercion" syntax with:
+
+    @Input() get disabled(): boolean { return this._disabled; }
+    set disabled(value: boolean) { this._disabled = (value === '') || value; }
+    // ngAcceptInputType is a hint for the Angular compiler to know that the input
+    // can in fact be a boolean or a string
+    static ngAcceptInputType_disabled: boolean | string;
+
+This can now be removed as TypeScript support a different type for the getter/setter pair \o/. There are no migration or warning messages for this change, so you'll have to search for `ngAcceptInputType` yourself.
```
Comment (jnizet) [inline _posts/2021-11-03-what-is-new-angular-13.0.md:40]: support -> supports

File: _posts/2021-11-03-what-is-new-angular-13.0.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+## Forms
+
+In a near future, we should finally have strictly typed forms
+(see the Angular v14 section below).
+In the meantime, a small improvement landed:
+`status` and `statusChanges` are now strictly typed as `FormControlStatus` and `Observable<FormControlStatus>` instead of `string` and `Observable<any>` previously.
+`FormControlStatus` is a new union type of the various status values:
+`'VALID' | 'INVALID' | 'PENDING' | 'DISABLED'`.
+This is technically a breaking change,
+but except if you were doing weird things,
```
Comment (jnizet) [inline _posts/2021-11-03-what-is-new-angular-13.0.md:81]: but unless you were doing weird things

File: _posts/2021-11-03-what-is-new-angular-13.0.md [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+that should work fine.
+
+## Templates
+
+The `fullTemplateTypeCheck` compiler is now deprecated
```
Comment (jnizet) [inline _posts/2021-11-03-what-is-new-angular-13.0.md:86]: compiler -> compiler option

File: _posts/2021-11-03-what-is-new-angular-13.0.md [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+{% raw %}
+    {{ date | date: "shortTime":"+1200" }}
+{% endraw %}
+
+Angular v13 added the possibility to define the default timezone to use
+for the `date` pipe, instead of having to repeat it in every template.
+To do so, you'll have to use the special token `DATE_PIPE_DEFAULT_TIMEZONE`:
+
+    providers: [
+      { provide: DATE_PIPE_DEFAULT_TIMEZONE, useValue: '+1200' }
```
Comment (jnizet) [inline _posts/2021-11-03-what-is-new-angular-13.0.md:109]: Aaaarg, it's not a really time zone, it's a time offset.

File: _posts/2021-11-03-what-is-new-angular-13.0.md [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+    ]
+
+## Core with factory-less APIs
+
+The `createComponent` API has been simplified to accept
+a component class directly instead of a component factory.
+Until now, you had to use this pretty terrible API:
```
Comment (jnizet) [inline _posts/2021-11-03-what-is-new-angular-13.0.md:116]: API to create a new component instance dynamically

File: _posts/2021-11-03-what-is-new-angular-13.0.md [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+hence making sure that the TypeScript compilation is not slowing down the feedback loop.
+In most cases, everything is fine with our code,
+and we just want to see the result in the browser as fast as possible.
+We would still have compilation errors if something is broken, just a bit later.
+One idea would be to leverage the awesome [Vite](https://vitejs.dev/) tooling,
+which does that, and much more.
+Vite is the new "hot" tool in the JS world, with Vue, Svelte, and React developers already using it.
+Without digging too much into the details,
+one of the strengths of Vite is to leverage ESM modules
+to give a way faster reloading experience while developing.
```
Comment (jnizet) [inline _posts/2021-11-03-what-is-new-angular-13.0.md:217]: way faster -> much faster

**Review by jnizet — CHANGES_REQUESTED** [review]

File: _posts/2021-11-03-what-is-new-angular-13.0.md (hunk 7 above)
Comment (cexbrayat) [reply _posts/2021-11-03-what-is-new-angular-13.0.md:109]: It can also be a standard timezone in addition to a timezone offset. I'll add a note.

**Pull request merged**